spec.html

```diff
@@ -35068,85 +35064,82 @@ <h1>Runtime Semantics: AllocateTypedArray ( _constructorName_, _newTarget_, _def
+            1. Assert: _O_ is an Object that has a [[TypedArrayName]] internal slot.
+            1. Assert: _buffer_ is an Object that has an [[ArrayBufferData]] internal slot.
+            1. Let _constructorName_ be the String value of _O_.[[TypedArrayName]].
+            1. Let _elementSize_ be the Element Size value specified in <emu-xref href="#table-the-typedarray-constructors"></emu-xref> for _constructorName_.
+            1. Let _offset_ be ? ToIndex(_byteOffset_).
+            1. If _offset_ modulo _elementSize_ &ne; 0, throw a *RangeError* exception.
+            1. If _length_ is not *undefined*, then
+              1. Let _newLength_ be ? ToIndex(_length_).
+            1. If IsDetachedBuffer(_buffer_) is *true*, throw a *TypeError* exception.
+            1. Let _bufferByteLength_ be _buffer_.[[ArrayBufferByteLength]].
+            1. If _length_ is *undefined*, then
+              1. If _bufferByteLength_ modulo _elementSize_ &ne; 0, throw a *RangeError* exception.
+              1. Let _newByteLength_ be _bufferByteLength_ - _offset_.
+              1. If _newByteLength_ &lt; 0, throw a *RangeError* exception.
+            1. Else,
+              1. Let _newByteLength_ be _newLength_ &times; _elementSize_.
+              1. If _offset_ + _newByteLength_ &gt; _bufferByteLength_, throw a *RangeError* exception.
+            1. Set _O_.[[ViewedArrayBuffer]] to _buffer_.
+            1. Set _O_.[[ByteLength]] to _newByteLength_.
+            1. Set _O_.[[ByteOffset]] to _offset_.
+            1. Set _O_.[[ArrayLength]] to _newByteLength_ / _elementSize_.
+          </emu-alg>
+        </emu-clause>
 
-      <emu-clause id="sec-typedarray-object">
-        <h1>_TypedArray_ ( _object_ )</h1>
-        <p>This description applies only if the _TypedArray_ function is called with at least one argument and the Type of the first argument is Object and that object does not have either a [[TypedArrayName]] or an [[ArrayBufferData]] internal slot.</p>
-        <p>_TypedArray_ called with argument _object_ performs the following steps:</p>
-        <emu-alg>
-          1. Assert: Type(_object_) is Object and _object_ does not have either a [[TypedArrayName]] or an [[ArrayBufferData]] internal slot.
-          1. If NewTarget is *undefined*, throw a *TypeError* exception.
-          1. Let _constructorName_ be the String value of the Constructor Name value specified in <emu-xref href="#table-the-typedarray-constructors"></emu-xref> for this <var>TypedArray</var> constructor.
-          1. Let _O_ be ? AllocateTypedArray(_constructorName_, NewTarget, <code>"%<var>TypedArray</var>.prototype%"</code>).
-          1. Let _usingIterator_ be ? GetMethod(_object_, @@iterator).
-          1. If _usingIterator_ is not *undefined*, then
-            1. Let _values_ be ? IterableToList(_object_, _usingIterator_).
+        <emu-clause id="sec-initializetypedarrayfromlist" aoid="InitializeTypedArrayFromList" oldids="sec-typedarray-object">
```

Comment:
Nit: since this only captures part of what the old `sec-typedarray-object` did, it probably makes more sense for that oldid to be attached to the actual constructor.

Reply:
It's a sticky case. To me, it makes more sense where it is, but I'll move it if any other editor agrees.

Reply:
We chatted about this briefly on the editor call and agree with @bakkot's preference here. The thinking is that for someone following an old link, it's easier to drill down here than to have to go back up, so let's link to the actual constructor.

Reply:
Done!